File: C_Programming/Practise_Programs/PP_107.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_107
//
//  Description       : This program demonstrates two different
//                      methods to calculate the length of a string,
//                      using strlen() and a character pointer.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 13/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<string.h>

int main()
{
    char *str = "Ganesh";    
    int iCount = 0; 
    
    printf("Length of string is : %lu\n", strlen(str));

    while(*str != '\0')
    {
        iCount++;
        str++;
    }    

    printf("Length of string is : %d\n", iCount);
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      Length of string is : 6
//      Length of string is : 6
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_107.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_107
//
//  Description       : This program accepts a string from the user
//                      and displays the entered string.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 26/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class PP_107
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String Arr = null;

        System.out.println("Enter String : ");
        Arr = sobj.nextLine();

        System.out.println("Entered String : " + Arr);
        
        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Jay Ganesh
//
//  Sample Output :
//
//      Enter String :
//      Entered String : Jay Ganesh
//
//////////////////////////////////////////////////////////////////